robottelo/hosts.py

```diff
@@ -52,6 +53,23 @@ class ContentHostError(Exception):
 
 class ContentHost(Host):
     run = Host.execute
+    subscribed = False
+
+    @cached_property
+    def nailgun_host(self):
+        """If this host is subscribed, provide access ot its nailgun object"""
+        if self.subscribed:
```

Comment:
Since this depends on whether or not its subscribed, I wouldn't cache the property.

If its referenced before the host is subscribed, then it'll get cached to `None` and its on the caller to delete it before calling it again - making the caller have to check `host.subscribed` anyway.